src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,72 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { setCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  const { id, title, completed, userId } = currentTodo ?? {};
+
+  useEffect(() => {
+    setLoading(true);
+
+    if (userId) {
+      getUser(userId)
+        .then(setUser)
+        .catch(e => e)
```

Comment:
The error handling in the promise chain is not effective. The catch block currently just swallows the error by returning it. Consider logging the error or displaying an error message to the user.